hosts/agents-common.nix

```diff
@@ -0,0 +1,148 @@
+# SPDX-FileCopyrightText: 2022-2024 TII (SSRC) and the Ghaf contributors
+# SPDX-License-Identifier: Apache-2.0
+
+{ pkgs, inputs, ... }:
+let
+  # Vendored in, as brainstem isn't suitable for nixpkgs packaging upstream:
+  # https://github.com/NixOS/nixpkgs/pull/313643
+  brainstem = pkgs.callPackage ../pkgs/brainstem { };
+in
+{
+  services.udev.packages = [
+    brainstem
+    pkgs.usbsdmux
+  ];
+
+  environment.systemPackages =
+    [
+      inputs.robot-framework.packages.${pkgs.system}.ghaf-robot
+      brainstem
+    ]
+    ++ (with pkgs; [
+      minicom
+      usbsdmux
+    ]);
+
+  # The Jenkins slave service is very barebones
+  # it only installs java and sets up jenkins user
+  services.jenkinsSlave.enable = true;
+
+  # Jenkins needs sudo and serial rights to perform the HW tests
+  users.users.jenkins.extraGroups = [
+    "wheel"
+    "dialout"
+    "tty"
+  ];
+
+  systemd.services =
+    let
+      # verifies that the environment file is properly configured
+      # and downloads agent.jar from the controller
+      jenkins-setup-script = pkgs.writeShellScript "jenkins-setup.sh" ''
+        if [[ ! -f jenkins.env ]]; then
+          echo "Please add jenkins controller details to $(pwd)/jenkins.env"
+          echo "CONTROLLER=" > jenkins.env
+          echo "ADMIN_PASSWORD=" >> jenkins.env
+          exit 1
```

Comment:
After this, `/var/lib/jenkins/jenkins.env` is world-readable. We should create that file so only the owner has read and write access (600).

Reply:
fixed